Apps/WebClient/src/ClientApp/src/components/modal/addQuickLink.vue

```diff
@@ -0,0 +1,321 @@
+<script lang="ts">
+import Vue from "vue";
+import { Component, Emit } from "vue-property-decorator";
+import { Action, Getter } from "vuex-class";
+
+import type { WebClientConfiguration } from "@/models/configData";
+import { QuickLink } from "@/models/quickLink";
+import { EntryType } from "@/models/timelineEntry";
+import User from "@/models/user";
+import { SERVICE_IDENTIFIER } from "@/plugins/inversify";
+import container from "@/plugins/inversify.container";
+import { ClientModule } from "@/router";
+import { ILogger } from "@/services/interfaces";
+
+interface QuickLinkFilter {
+    name: string;
+    module: string;
+    isEnabled: boolean;
+}
+
+@Component({
+    components: {},
+})
+export default class AddQuickLinkComponent extends Vue {
+    @Getter("user", { namespace: "user" }) user!: User;
+
+    @Getter("webClient", { namespace: "config" })
+    webClientConfig!: WebClientConfiguration;
+
+    @Action("updateQuickLinks", { namespace: "user" })
+    updateQuickLinks!: (params: {
+        hdid: string;
+        quickLinks: QuickLink[];
+    }) => void;
+
+    @Getter("quickLinks", { namespace: "user" }) quickLinks!:
+        | QuickLink[]
+        | undefined;
+
+    private logger!: ILogger;
+    private checkboxComponentKey = 0;
+    private isVisible = false;
+
+    private selectedQuickLinks: string[] = [];
+    private enabledQuickLinks: QuickLinkFilter[] = [];
+
+    private get quickLinkFilter(): QuickLinkFilter[] {
+        return [
+            {
+                name: "Health Visits",
+                module: ClientModule.Encounter,
+                isEnabled: this.webClientConfig.modules[EntryType.Encounter],
+            },
+            {
+                name: "COVID-19 Tests",
+                module: ClientModule.Laboratory,
+                isEnabled:
+                    this.webClientConfig.modules[
+                        EntryType.Covid19LaboratoryOrder
+                    ],
+            },
+            {
+                name: "My Notes",
+                module: ClientModule.Note,
+                isEnabled: this.webClientConfig.modules[EntryType.Note],
+            },
+            {
+                name: "Special Authority",
+                module: ClientModule.MedicationRequest,
+                isEnabled:
+                    this.webClientConfig.modules[EntryType.MedicationRequest],
+            },
+        ];
+    }
+
+    private get enabledQuickLinkFilter(): QuickLinkFilter[] {
```

Comment:
This could probably be simplified to return a filtered array without needing to call other functions.

Reply:
Simplified as store getter bug has been fixed.